src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -273,15 +279,16 @@ public async Task<AnalysisResult> GetAnalysisResultAsync(ImmutableArray<Diagnost
         /// <summary>
         /// Returns all diagnostics produced by compilation and by all <see cref="Analyzers"/>.
         /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public Task<ImmutableArray<Diagnostic>> GetAllDiagnosticsAsync()
         {
-            return GetAllDiagnosticsAsync(_cancellationToken);
+            return GetAllDiagnosticsAsync(cancellationToken: default);
         }
 
         /// <summary>
         /// Returns all diagnostics produced by compilation and by all <see cref="Analyzers"/>.
         /// </summary>
-        public async Task<ImmutableArray<Diagnostic>> GetAllDiagnosticsAsync(CancellationToken cancellationToken)
+        public async Task<ImmutableArray<Diagnostic>> GetAllDiagnosticsAsync(CancellationToken cancellationToken = default)
```

Comment:
see above.  This will be fine.  from the user perspective, they'll see a single method with an optional CT (like all of our compiler CT-taking methods).  If they call it with a CT this gets calls.  If they call it without a CT, technically anotehr method is called.  But that method calls right back into this, so the net effect is the exact same for the user.